### PR TITLE
Remove Deployment Manager from build/

### PR DESCRIPTION
**Pull request by markmandel**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind breaking
> /kind bug

/kind cleanup

> /kind documentation
> /kind feature
> /kind hotfix

**What this PR does / Why we need it**:

This removes the final pieces of Deployment Manager from our build tooling.

* Switches e2e cluster creation and deletion to terraform
* Switches prow cluster creation and deletion to terraform
* Fixes a small issue with `clean-gcloud-test-cluster`
* Updated development README to match the new functionality.

Agones Developers are recommended to manually deleted the previous deployment through the Google Cloud console before applying these Terraform configurations.

**Which issue(s) this PR fixes**:
N/A

**Special notes for your reviewer**:
N/A

### CODE REVIEW

File: build/includes/google-cloud.mk [hunk 1]
```diff
@@ -31,39 +31,32 @@ gcloud-test-cluster: $(ensure-build-image) terraform-init
...
+	docker run --rm -it $(common_mounts) $(DOCKER_RUN_ARGS) $(build_tag) bash -c 'cd $(mount_path)/build/terraform/e2e && \
+      	terraform init && gcloud auth application-default login && terraform apply -auto-approve -var project="$(GCP_PROJECT)"'
 
 # Deletes the gcloud e2e cluster and cleanup any left pvc volumes
-clean-gcloud-e2e-test-cluster: TEST_CLUSTER_NAME ?= e2e-test-cluster
-clean-gcloud-e2e-test-cluster: $(ensure-build-image)
-	-docker run --rm $(common_mounts) $(DOCKER_RUN_ARGS) $(build_tag) \
-		helm uninstall consul && kubectl delete pvc -l component=consul-consul
-	$(MAKE) clean-gcloud-test-cluster GCP_CLUSTER_NAME=$(TEST_CLUSTER_NAME)
+clean-gcloud-e2e-test-cluster: $(ensure-build-image) terraform-init
```
Comment (roberthbailey) [inline build/includes/google-cloud.mk:46]: qq: why have the "clean" targets depend on `terraform-init` whereas the "create" targets do it inline in the docker run command?

**Comment by roberthbailey** [comment]
Also, should we hold this during the feature freeze or not worry about it since it's changing dev tooling?

**Comment by markmandel** [comment]
I think it's fine to go through, since it's dev tooling, and doesn't impact the end-user at all.

But happy to be dissuaded by dissenting opinions.

**Comment by roberthbailey** [comment]
I just want to make it clear on why we would allow it so there isn't any concern of bias (e.g. we let Mark's PRs go in during freeze).

**Comment by markmandel** [comment]
That's a good point, our [release management](https://github.com/googleforgames/agones/blob/master/docs/governance/release_process.md) script does say:

> For the next week, the project is in "feature freeze" - i.e. only bug and documentation (.md and examples) fixes during this time.

Should this be extended to "build tools" as well?

File: build/includes/google-cloud.mk (hunk 1 above)
Comment (markmandel) [reply build/includes/google-cloud.mk:46]: OH! you are right! I'm double init'ing - let me clean that up!

Comment (markmandel) [reply build/includes/google-cloud.mk]: Fixed this all up - it was all types of messed actually, so please re-review.

**Pull request merged**